src/components/plantCollection/PlantCollection.tsx

```diff
@@ -1,32 +1,42 @@
 import NavItem from "../navItem/NavItem.tsx";
 import { PlantCollectionStyled } from "./PlantCollection.styles.ts";
-import { useEffect, useState } from "react";
 import { pb, PLANTS_COLLECTION } from "../../Backend.constants.ts";
-import { FormValues } from "../form/Form.types.ts";
+import { useQuery } from "@tanstack/react-query";
+import Text from "../text/Text.tsx";
 
 const PlantCollection = () => {
-  const [plantsCollection, setPlantsCollection] = useState<FormValues[] | []>(
-    [],
-  );
+  const getPlantsList = async () => {
+    const res = await pb.collection(PLANTS_COLLECTION).getList(1, 100);
 
-  useEffect(() => {
-    pb.collection(PLANTS_COLLECTION)
-      .getList<FormValues>(1, 100)
-      .then((result) => setPlantsCollection(result.items));
-    console.log(plantsCollection);
-  }, []);
+    return res.items;
+  };
 
-  return (
-    <PlantCollectionStyled>
-      {plantsCollection.map((item) => {
-        return (
-          <NavItem key={item.id} linkTo={`/${item.plantName}`}>
-            {item.plantName}
-          </NavItem>
-        );
-      })}
-    </PlantCollectionStyled>
-  );
+  const { isPending, isError, data, error } = useQuery({
+    queryKey: ["plants"],
+    queryFn: getPlantsList,
+  });
+
+  if (isPending) {
+    return <Text variant={"large"}>Loading...</Text>;
+  }
+
+  if (isError) {
+    return <Text variant={"large"}>Error: {error.message}</Text>;
+  }
+
+  if (data) {
+    return (
+      <PlantCollectionStyled>
+        {data.map((plant) => {
```

Comment:
redundant brackets

Reply:
you mean around `(plant)`? I've tried to remove them but my code editor brings them back😅

Reply:
after fat arrow :) you don't need an explicit `return` here